src/commands/addMIConnections/addLocalMIConnections.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AppSettingTreeItem } from "@microsoft/vscode-azext-azureappsettings";
+import { AzureWizard, type AzureWizardExecuteStep, type AzureWizardPromptStep, type IActionContext } from "@microsoft/vscode-azext-utils";
+import { type MessageItem } from "vscode";
+import { localize } from "../../localize";
+import { createActivityContext } from "../../utils/activityUtils";
+import { ConnectionsListStep, type Connection } from "./ConnectionsListStep";
+import { type IAddMIConnectionsContext } from "./IAddMIConnectionsContext";
+import { LocalSettingsAddStep } from "./LocalSettingsAddStep";
+import { SettingsAddBaseStep } from "./SettingsAddBaseStep";
+
+export async function addLocalMIConnections(context: IActionContext, node?: AppSettingTreeItem): Promise<void> {
+    const connections: Connection[] = [];
+    if (node instanceof AppSettingTreeItem) {
+        connections.push({ name: node.id, value: node.value });
+        await node.parent.runWithTemporaryDescription(context, localize('adding', 'Adding...'), async () => {
+            await addLocalMIConnectionsInternal(context, connections);
+        });
+    } else {
+        await addLocalMIConnectionsInternal(context);
```

Comment:
If you care enough, you can use the `ICreateChildImplContext` to show a "Creating..." item in the tree while the create is in progress. Not sure how long it takes, I imagine not that long.